app/controllers/pensions/paymentsIntoOverseasPensions/PaymentsIntoOverseasPensionsCYAController.scala

```diff
@@ -41,43 +39,37 @@ import scala.concurrent.{ExecutionContext, Future}
 class PaymentsIntoOverseasPensionsCYAController @Inject() (auditProvider: AuditActionsProvider,
                                                            view: PaymentsIntoOverseasPensionsCYAView,
                                                            errorHandler: ErrorHandler,
-                                                           service: PaymentsIntoOverseasPensionsService,
-                                                           mcc: MessagesControllerComponents)(implicit appConfig: AppConfig, ec: ExecutionContext)
+                                                           pensionsService: PensionsService,
+                                                           mcc: MessagesControllerComponents)(implicit appConfig: AppConfig)
     extends FrontendController(mcc)
     with I18nSupport
-    with SessionHelper {
+    with Logging {
+
+  implicit val executionContext: ExecutionContext = mcc.executionContext
 
   def show(taxYear: TaxYear): Action[AnyContent] = auditProvider.paymentsIntoOverseasPensionsViewAuditing(taxYear) async { implicit request =>
     val cyaData    = request.sessionData
     val taxYearInt = taxYear.endYear
 
     if (cyaData.pensions.paymentsIntoOverseasPensions.isFinished) {
-      Future.successful(Ok(view(taxYearInt, cyaData.pensions.paymentsIntoOverseasPensions)))
+      Future.successful(Ok(view(taxYear, cyaData.pensions.paymentsIntoOverseasPensions)))
     } else {
       val checkRedirect = journeyCheck(PaymentsIntoOverseasPensionsCYAPage, _: PensionsCYAModel, taxYearInt)
       redirectBasedOnCurrentAnswers(taxYearInt, Some(cyaData), cyaPageCall(taxYearInt))(checkRedirect) { data =>
-        Future.successful(Ok(view(taxYearInt, data.pensions.paymentsIntoOverseasPensions)))
+        Future.successful(Ok(view(taxYear, data.pensions.paymentsIntoOverseasPensions)))
       }
     }
   }
 
-  def submit(taxYear: Int): Action[AnyContent] = auditProvider.paymentsIntoOverseasPensionsUpdateAuditing(taxYear) async { implicit request =>
-    val checkRedirect = journeyCheck(PaymentsIntoOverseasPensionsCYAPage, _: PensionsCYAModel, taxYear)
-    redirectBasedOnCurrentAnswers(taxYear, Some(request.sessionData), cyaPageCall(taxYear))(checkRedirect) { sessionData =>
-      if (sessionDataDifferentThanPriorData(sessionData.pensions, request.maybePrior)) {
+  def submit(taxYear: TaxYear): Action[AnyContent] = auditProvider.paymentsIntoOverseasPensionsUpdateAuditing(taxYear.endYear) async {
+    implicit request =>
+      val res = pensionsService.upsertPaymentsIntoPensions(
```

Comment:
this will need to be `pensionsService.upsertPaymentsIntoOverseasPensions` instead of `pensionsService.upsertPaymentsIntoPensions` so will need a new method in service and connector that matches the format of the existing ones for other journeys

Reply:
Thanks, I've pushed my changes, please take a look and let me know if they are ok.